apps/frontend/src/components/TimeCardPage/TimeSheet.tsx

```diff
@@ -1,341 +1,415 @@
+
+  // fetch the information of the user whos timesheet is being displayed
+  // if user is an employee selected and user would be the same
+  // if user is a supervisor/admin then selected would contain the information of the user
+  // whos timesheet is being looked at and user would contain the supervisor/admins information
+  // by default the first user is selected
+  const [selectedUser, setSelectedUser] = useState<UserSchema>();
+  const [user, setUser] = useState<UserSchema>();
+
+  // associates is only used by supervisor/admin for the list of all associates they have access to
+  const [associates, setAssociates] = useState<UserSchema[]>([]);
+
+  // A list of the timesheet objects
+  // TODO: add types
+  const [userTimesheets, setUserTimesheets] = useState([]);
+  const [currentTimesheets, setCurrentTimesheets] = useState([]);
+  const [selectedTimesheet, setTimesheet] = useState(undefined);
+
+  const [weeklyComments, setWeeklyComments] = useState<CommentSchema[]>([]);
+  const [weeklyReports, setWeeklyReports] = useState<CommentSchema[]>([]);
+
+  // this hook should always run first
+  useEffect(() => {
+    // Get the current logged in user
+    getCurrentUser().then((currentUser) => {
+      console.log("Current user: ", currentUser);
+      setUser(currentUser);
+    });
+
+    apiClient.getUser().then((userInfo) => {
+      if (userInfo.Type === "Supervisor" || userInfo.Type === "Admin") {
+        apiClient.getAllUsers().then((users) => {
+          setAssociates(users);
+          setSelectedUser(users[0]);
         });
-        setUserTimesheets(modifiedTimesheets);
-
-        //Also need to update our list of currently selected - TODO come up with a way to not need these duplicated lists
-        setCurrentTimesheets(currentTimesheets.map(
-            (entry) => {
-                if (entry.TimesheetID === selectedTimesheet.TimesheetID) {
-                    return {
-                        ...entry,
-                        TableData: updated_sheet.TableData
-                    }
-                }
-                return entry
-            }
-        ));
+      }
+      setSelectedUser(userInfo);
+    });
+    // if employee setSelectedUSer to be userinfo
+    // if supervisor/admin get all users
+    // set selected user
+  }, []);
+
+  const getUpdatedTimesheet = (userId) => {
+    apiClient.getUserTimesheets(userId).then((timesheets) => {
+      setUserTimesheets(timesheets);
+      //By Default just render / select the first timesheet for now
+      setCurrentTimesheetsToDisplay(timesheets, selectedDate);
+    });
+  };
+
+  // Pulls user timesheets, marking first returned as the active one
+  useEffect(() => {
+    getUpdatedTimesheet(selectedUser?.UserID);
+  }, [selectedUser]);
+
+  // Callback function that triggers GET request to the API to grab the most recent version of timesheets for the current selected user,
+  // and re-sets the current timesheet state variable
+  const forceRefreshTimesheet = () => {
```

Comment:
@circlegov also these functions which are the new callback function to force refresh data